### ISSUE TITLE
Various storage capacity outputs

### ISSUE THREAD

**Issue by caro-rli** [oemof/oemof#152]
I have found a funny bug.. or at least something I cannot explain. Running the [storage_invest.py](oemof/examples/storage_optimization/storage_invest.py) in the current dev leads to different outputs of

``` python
print(energysystem.results[storage].add_cap)
```

Outputs of the last eight simulation runs have been:

```
10965770.150517
10973781.8191967
10969130.4703331
10969130.4703331
10969130.4703331
10965770.150517
10969817.8760405
10969130.4703331
```

Going back to v0.0.3 is even worse. Here the outputs vary such as this:

```
11070491.4073288
11105431.4830159
11041116.6957001
11105431.4830159
11041116.6957001
11105431.4830159
```

This might be not a big issue, but at least needs to be explained.

And we might get trouble when creating a results check as proposed in #149

**Comment by caro-rli** [comment]
Every other output seems to be the same in each simulation run.

**Comment by caro-rli** [comment]
@gnn might this have something to do with #58 ?

**Comment by simnh** [comment]
Could you post the solver output files ? Handle very large coefficient ranges can cause numerical instability for solvers, thus resulting in increasing tolerances. 
That would be my point to start...

**Comment by simnh** [comment]
There might be another reason, which is higly unlikely. If your problem is degenerated and row in the matrix are somehow mixed up from one run to another the results can differ. This you be checked firstly be looking at the lp files and order of constraints and variables. But pyomo should actually take care of the same order and the solver of information about degeneracy too. So we should start looking somewhere else first.

**Comment by simnh** [comment]
are there differences in the objective value as well? (probably i guess)

**Comment by SanderNielen** [comment]
I noticed an issue that might be related. When I have an energy system with a very efficient storage (100% for both charge and discharge) that lacks `opex_var`, the storage is constantly exchanging energy with itself. This dynamic equilibrium can be situated at a random (dis)charge rate. The equilibrium is disrupted when the component actually needs to store or release energy. 
A possible solution would be to add variable OPEX or a lower efficiency. But perhaps there is a better way?

**Comment by uvchik** [comment]
The reason is, that the solver does not think like a human being. If you make a cost optimisation and loading and unloading the storage a hundred times is not more expansive than not doing it, it is still one of the cheapest solution.

The reason why we think, that we should not use a storage like this is that there are cost, losses and abrasion. So we have to tell the solver about these things. If the storage would be as perfect as in your model we wouldn't care about this behaviour.

@caro-rli: Do you still have these problems? I figured out that there are different values using the glpk-solver but not using the cbc-solver. Could you test it using the cbc?

**Comment by ckaldemeyer** [comment]
> @SanderNielen: The reason is, that the solver does not think like a human being. If you make a cost optimisation and loading and unloading the storage a hundred times is not more expansive than not doing it, it is still one of the cheapest solution.

> The reason why we think, that we should not use a storage like this is that there are costs, losses and abrasions. So we have to tell the solver about these things. If the storage would be as perfect as in your model we wouldn't care about this behaviour.

I agree to @uvchik . On workarround to get rid of the behaviour would be to add very (not noticeable) low losses.

**Comment by caro-rli** [comment]
I think we solved this with #160 .
